models/repo.go

```diff
@@ -1830,10 +1840,22 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos []*Repository, _ int
 		return nil, 0, fmt.Errorf("Count: %v", err)
 	}
 
-	return repos, count, sess.
+	if err = sess.
 		Limit(opts.PageSize, (opts.Page-1)*opts.PageSize).
 		OrderBy(opts.OrderBy).
-		Find(&repos)
+		Find(&repos); err != nil {
+		return nil, 0, fmt.Errorf("Repo: %v", err)
+	}
+
+	if opts.Starred {
```

Comment:
Maybe you could see https://github.com/go-gitea/gitea/blob/master/models/repo.go#L2156